libafl/src/executors/forkserver.rs

```diff
@@ -0,0 +1,417 @@
+
+        //setsid, setrlimit, direct stdin, set pipe, and finally fork.
+        match Command::new(target)
+            .args(args)
+            .stdin(Stdio::null())
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .env("LD_BIND_LAZY", "1")
+            .setlimit(memlimit)
+            .setsid()
+            .setstdin(out_filefd, use_stdin)
+            .setpipe(
+                st_pipe.read_end().unwrap(),
+                st_pipe.write_end().unwrap(),
+                ctl_pipe.read_end().unwrap(),
+                ctl_pipe.write_end().unwrap(),
+            )
+            .spawn()
+        {
+            Ok(_) => {}
+            Err(_) => {
+                panic!("Command::new() failed");
+            }
+        };
+
+        //parent: we'll close unneeded endpoints
+        ctl_pipe.close_read_end();
+        st_pipe.close_write_end();
+
+        Self {
+            st_pipe: st_pipe,
+            ctl_pipe: ctl_pipe,
+            child_pid: 0,
+            status: 0,
+            last_run_timed_out: 0,
+        }
+    }
+
+    pub fn last_run_timed_out(&self) -> i32 {
+        self.last_run_timed_out
+    }
+
+    pub fn status(&self) -> i32 {
+        self.status
+    }
+
+    pub fn child_pid(&self) -> u32 {
+        self.child_pid
+    }
+
+    pub fn read_st(&mut self) -> Result<(usize, i32), io::Error> {
+        let mut buf: [u8; 4] = [0u8; 4];
+
+        let rlen = self.st_pipe.read(&mut buf)?;
+        let val: i32 = i32::from_ne_bytes(buf);
+
+        Ok((rlen, val))
+    }
+
+    pub fn write_ctl(&mut self, val: i32) -> Result<usize, io::Error> {
+        let buf = val.to_ne_bytes();
+        let slen = self.ctl_pipe.write(&buf)?;
+
+        Ok(slen)
+    }
+}
+
+pub struct ForkserverExecutor<I, OC, OF, OT, S>
+where
+    I: Input + HasTargetBytes,
+    OC: Corpus<I>,
+    OF: Feedback<I, S>,
+    OT: ObserversTuple,
+{
+    target: String,
+    args: Vec<String>,
+    out_file: OutFile,
+    forkserver: Forkserver,
+    observers: OT,
+    phantom: PhantomData<(OC, OF, I, S)>,
+}
+
+impl<I, OC, OF, OT, S> ForkserverExecutor<I, OC, OF, OT, S>
+where
+    I: Input + HasTargetBytes,
+    OC: Corpus<I>,
+    OF: Feedback<I, S>,
+    OT: ObserversTuple,
+{
+    pub fn new(target: String, argv: Vec<String>, observers: OT) -> Result<Self, Error> {
+        let mut args = Vec::<String>::new();
+        let mut use_stdin = true;
+        let out_filename = format!("out-{}", 123456789); //TODO: replace it with a random number
+
+        for item in argv {
+            if item == "@@" && use_stdin {
+                use_stdin = false; //only 1 '@@' allowed.
+                args.push(out_filename.clone());
+            } else {
+                args.push(item.to_string());
+            }
+        }
+
+        let args: Vec<String> = args.iter().map(|s| s.to_string()).collect();
+
+        let out_file = OutFile::new(&out_filename);
+
+        //forkserver
```

Comment:
This and other comments are useless. When you write comments try to do not add the obvious, and use this format:
`// Comment` which has a space after `//` and start with a capital letter.